diff --git a/launcher/minecraft/ShortcutUtils.cpp b/launcher/minecraft/ShortcutUtils.cpp
index ec4ebb31a..7764bf44d 100644
--- a/launcher/minecraft/ShortcutUtils.cpp
+++ b/launcher/minecraft/ShortcutUtils.cpp
@@ -98,6 +98,7 @@ bool createInstanceShortcut(const Shortcut& shortcut, const QString& filePath)
     }
 
     iconPath = FS::PathCombine(shortcut.instance->instanceRoot(), "icon.png");
+    appPath = "/snap/bin/prismlauncher-alpo.prismlauncher";
 
     QFile iconFile(iconPath);
     if (!iconFile.open(QFile::WriteOnly)) {
