diff --git a/launcher/minecraft/ShortcutUtils.cpp b/launcher/minecraft/ShortcutUtils.cpp
index 7d662fe8b..6918cac93 100644
--- a/launcher/minecraft/ShortcutUtils.cpp
+++ b/launcher/minecraft/ShortcutUtils.cpp
@@ -97,6 +97,7 @@ bool createInstanceShortcut(const Shortcut& shortcut, const QString& filePath)
         }
     }
 
+    appPath = "/snap/bin/prismlauncher-alpo.prismlauncher";
     iconPath = FS::PathCombine(shortcut.instance->instanceRoot(), "icon.png");
 
     QFile iconFile(iconPath);
diff --git a/launcher/ui/dialogs/CreateShortcutDialog.cpp b/launcher/ui/dialogs/CreateShortcutDialog.cpp
index 7d4199f13..c2aa07ccc 100644
--- a/launcher/ui/dialogs/CreateShortcutDialog.cpp
+++ b/launcher/ui/dialogs/CreateShortcutDialog.cpp
@@ -215,8 +215,6 @@ void CreateShortcutDialog::createShortcut()
     ShortcutUtils::Shortcut args{ m_instance, name, targetString, this, extraArgs, InstIconKey, target };
     if (target == ShortcutTarget::Desktop)
         ShortcutUtils::createInstanceShortcutOnDesktop(args);
-    else if (target == ShortcutTarget::Applications)
-        ShortcutUtils::createInstanceShortcutInApplications(args);
     else
         ShortcutUtils::createInstanceShortcutInOther(args);
 }
