diff --git a/src/wl_window.c b/src/wl_window.c
index 0e67e5c5..cddcac44 100644
--- a/src/wl_window.c
+++ b/src/wl_window.c
@@ -2230,8 +2230,8 @@ void _glfwSetWindowIconWayland(_GLFWwindow* window,
 {
     if (!_glfw.wl.toplevelIconManager)
     {   
-        _glfwInputError(GLFW_FEATURE_UNAVAILABLE,
-                        "Wayland: The platform does not support setting the window icon");
+        fprintf(stderr,
+                    "[GLFW] Wayland: The compositor does not support setting the window icon");
         return;
     }
 
--

